src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -542,9 +542,9 @@ def update_existing_check_in(
             # 04
             # Update monitor status
             if check_in.status == CheckInStatus.ERROR:
-                mark_failed(check_in, ts=start_time)
+                mark_failed(check_in, ts=check_in.date_added)
             else:
-                mark_ok(check_in, ts=start_time)
+                mark_ok(check_in, ts=check_in.date_added)
```

Comment:
there could be some 1 run weirdness here for overlapping monitors where `last_checkin__gt` evaluates to `False`
https://github.com/getsentry/sentry/blob/0c49a0036cf718d968a3fe02d7595d3c97358708/src/sentry/monitors/logic/mark_ok.py#L37

Reply:
same goes for `mark_failed`

Reply:
we would skip the `mark` call as the most recent check-in would be considered "earlier" than the previous check-in, but it would rectify on the next check-in.

we should add some overlapping check-in tests on our endpoints also just in general